pkg/importer/hcl2_test.go

```diff
@@ -506,6 +506,133 @@ func TestGenerateHCL2DefinitionsDoesNotMakeSelfReferences(t *testing.T) {
 	assert.Equal(t, expectedCode, hcl2Text.String(), "Generated HCL2 code does not match expected code")
 }
 
+func TestGenerateHCL2DefinitionsReplacesDeeplyNestedReferencesToLiterals(t *testing.T) {
```

Comment:
Can we have some test cases where there are multiple resources with the same output value, and therefore shouldn't be used as an input reference anywhere?

Reply:
`TestGenerateHCL2DefinitionsWithAmbiguousReferencesMaintainsLiteralValue` asserts just that but with IDs which are treated like outputs.